web-server/src/components/Settings/SyncDaysSetting.tsx

```diff
@@ -0,0 +1,146 @@
+    const res = await dispatch(
+      updateDefaultSyncDaysSettings({ orgId, defaultSyncDays: days })
+    );
+    return res.payload as number;
+  };
+
+  const handleDiscard = () => {
+    setSyncDays(initialSyncDays);
+    setError('');
+  };
+
+  const handleSave = async () => {
+    try {
+      await updateSyncDays(syncDays);
+      enqueueSnackbar('Default Sync Days Updated Successfully', {
+        variant: 'success',
+        autoHideDuration: 3000
+      });
+    } catch (e) {
+      handleDiscard();
+      return enqueueSnackbar('Default Sync Days Update Failed', {
+        variant: 'error',
+        autoHideDuration: 2000
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (!orgId) return;
+    getSyncDays().then((d) => setSyncDays(d));
+  }, []);
+
+  return (
+    <FlexBox col gap2>
+      <FlexBox big bold gap1 component={Line} white sx={{ fontSize: '20px' }}>
+        Organization Settings
+      </FlexBox>
+
+      <Divider />
+      {initialSyncDays && (
+        <>
+          <FlexBox gap1 alignCenter justifyBetween>
+            <FlexBox col flexGrow={1}>
+              <Line big bold>
+                Sync Days
+              </Line>
+              <Box>
+                The setting to control how many days of past data should be
+                included.
+              </Box>
+            </FlexBox>
+
+            <FlexBox alignCenter>
+              <TextField
+                value={syncDays}
+                onChange={(e) => {
+                  const days = Number(e.target.value);
+                  if (isNaN(days)) setError('Sync days is Not a number');
+                  else if (days < initialSyncDays)
+                    setError('Less than Current Sync Days not allowed');
+                  else if (days > 366)
+                    setError('Sync days cannot be more than 366');
+                  else setError('');
+                  setSyncDays(days);
+                }}
+                placeholder="default sync days"
+                sx={{ minWidth: '240px' }}
+                autoComplete="off"
+                error={!!error}
+                label={error}
+              />
+            </FlexBox>
+          </FlexBox>
+          <FlexBox
+            sx={{
+              justifyContent: 'flex-end',
+              marginTop: '16px'
+            }}
+            gap2
+          >
+            <Button
+              variant="outlined"
+              disabled={isSaving}
+              sx={{
+                '&.Mui-disabled': {
+                  borderColor: 'secondary.light'
+                },
+                width: '160px'
+              }}
+              onClick={handleDiscard}
+            >
+              DISCARD
+            </Button>
+            <LoadingButton
+              type="submit"
+              variant="outlined"
+              color="primary"
+              loading={isSaving}
+              disabled={syncDays == initialSyncDays || !!error}
```

Comment:
why double-equals? is this not typesafe?

Reply:
Yes it is typesafe. Apologies.